pandas/core/series.py

```diff
@@ -700,6 +701,72 @@ def view(self, dtype=None):
 
     # ----------------------------------------------------------------------
     # NDArray Compat
+    _HANDLED_TYPES = (Index, ExtensionArray, np.ndarray, numbers.Number,
+                      list)  # what other builtins? array? deque? ...
+
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        # TODO: handle DataFrame
+        from pandas.core.internals.construction import extract_array
+
+        # for binary ops, use our custom dunder methods
+        result = ops.maybe_dispatch_ufunc_to_dunder_op(
+            self, ufunc, method, *inputs, **kwargs)
+        if result is not NotImplemented:
+            return result
+
+        # align all the inputs.
+        names = [getattr(x, 'name') for x in inputs if hasattr(x, 'name')]
+        types = tuple(type(x) for x in inputs)
+        # TODO: dataframe
+        alignable = [x for x, t in zip(inputs, types) if issubclass(t, Series)]
+
+        if len(alignable) > 1:
+            # This triggers alignment.
+            # TODO: I'm sure there's a better way to get the expected index.
+            index = alignable[0].index
+            for s in alignable[1:]:
+                index |= s.index
+            inputs = [x.reindex(index) for x, t in zip(inputs, types)
+                      if issubclass(t, Series)]
```

Comment:
should there be a `else x` in the end for if you have a mixture of alignable and non-alignable objects?

Reply:
Are there any ufuncs that have more than 2 inputs? This is under an `if len(alignables) > 1`, so it should only be hit when we have 2 series.

But yes, I'll add an `else x` there, even if it may not (currently) be hit.

Reply:
Ah, that's a good point :) 
Indeed they all only have one or two inputs (https://docs.scipy.org/doc/numpy/reference/ufuncs.html). 

A comment about that might be just as good then